src/rubrix/client/datasets.py

```diff
@@ -0,0 +1,780 @@
+#  coding=utf-8
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import logging
+from typing import Dict, List, Optional, Tuple, Type, Union
+
+import pandas as pd
+from pkg_resources import parse_version
+
+from rubrix.client.models import (
+    Record,
+    Text2TextRecord,
+    TextClassificationRecord,
+    TokenAttributions,
+    TokenClassificationRecord,
+)
+from rubrix.client.sdk.datasets.models import TaskType
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DatasetBase:
+    """The Dataset classes are containers for Rubrix records.
+
+    This is the base class to facilitate the implementation for each record type.
+
+    Args:
+        records: A list of Rubrix records.
+
+    Raises:
+        WrongRecordTypeError: When the record type in the provided
+            list does not correspond to the dataset type.
+    """
+
+    _RECORD_TYPE = None
+
+    def __init__(self, records: Optional[List[Record]] = None):
```

Comment:
As a future improvement, we should accept a lazy list of records

Reply:
This is one of the points in the list ...